Dockerfile

```diff
@@ -1,12 +1,17 @@
-FROM python
+FROM python:3.10-alpine
 
 WORKDIR /app/
 
-COPY scripts ./scripts
-COPY requirements.txt ./
-RUN /app/scripts/build
+COPY . /app/
+RUN pip install --upgrade pip
+
+# requirements.txt lists `cryptography` which needs `cffi`
+# which doesn't have a musl (Alpine) wheel and needs `gcc`
+# and libs for installation from source
+RUN apk add --update --no-cache --virtual .build-deps \
+      gcc musl-dev libffi-dev \
+    && /app/scripts/build \
+    && apk del .build-deps
```

Comment:
I understand the switch to apk, but I'm not sure I see the advantage of reordering the layers...? Weren't the layers better ordered before in terms of optimizing the usability of the build cache ?

Also, can we compare with `python:slim` which uses a debian slim OS ?

Reply:
For development it is easier to mount the source into the container, instead of rebuilding image every time to take advantage of the build cache. The image is so small, that I didn't even think about caches.

The layers are merged to single step, to avoid storing `apk` build cache in the layer history.

I never used `python:slim`, so I don't have opinion about that. I've tested the setup with Alpine and it is working correctly.